### PR TITLE
Frontend: start fixing lint warnings

### PR DESCRIPTION
**Pull request by BacLuc**
After
- #5170 
we can start with fixing the lint issues.
See the single commits for further details.

### CODE REVIEW

File: frontend/src/components/form/api/ApiWrapper.vue [hunk 1]
```diff
@@ -5,7 +5,7 @@ Wrapper component for form components to save data back to API
 <template>
   <ValidationObserver ref="validationObserver" v-slot="validationObserver" slim>
     <v-form
-      :class="[{ 'api-wrapper--inline': !autoSave && !readonly && !separateButtons }]"
```
Comment (usu) [inline frontend/src/components/form/api/ApiWrapper.vue]: why this change?

File: frontend/src/components/activity/DaySwitcher.vue [hunk 2]
```diff
@@ -115,15 +115,21 @@ export default {
...
 </script>
 
+<!-- these styles should apply to inner elements -->
+<!-- eslint-disable-next-line vue-scoped-css/enforce-style-type -->
 <style>
 .basis-num {
   flex-basis: 2.5ch;
 }
 
+/* .e-day-switcher__menu is in the <e-select> tag */
```
Comment (usu) [inline frontend/src/components/activity/DaySwitcher.vue:125]: shouldn't it then be a scoped deep selector?

File: frontend/src/components/dialog/DialogForm.vue [hunk 3]
```diff
@@ -133,7 +133,7 @@ export default {
 }
 </script>
 
-<style lang="scss">
+<style lang="scss" scoped>
 @media #{map-get($display-breakpoints, 'xs-only')} {
   .ec-dialog-form {
```
Comment (usu) [inline frontend/src/components/dialog/DialogForm.vue:138]: .ec-dialog-form seems also referenced in other components (`DetailPane`, `DialogBottomSheet`). We might need to double check if it's really needed in these other components.

File: frontend/src/components/form/api/ApiWrapper.vue (hunk 1 above)
Comment (BacLuc) [reply frontend/src/components/form/api/ApiWrapper.vue]: Ui, good that you caught that.
Was to Test that the Styles still apply

File: frontend/src/components/activity/DaySwitcher.vue (hunk 2 above)
Comment (BacLuc) [reply frontend/src/components/activity/DaySwitcher.vue:125]: you tell me.
I have not much experience in structuring (s)css or vue.
I just try to put scoped where the linter says i should (which is everywhere).
If this breaks something, i suppress the lint warning, and hope that someone with more experience really improves it in the future.

**Comment by manuelmeister** [comment]
I think I'll need to check each changed view. Maybe it would be sensible for me to refactor the unscoped components. It will probably still need some unscoped thematic css, and I'm not sure if we should keep this linter rule.

**Comment by BacLuc** [comment]
> I think I'll need to check each changed view. Maybe it would be sensible for me to refactor the unscoped components. It will probably still need some unscoped thematic css, and I'm not sure if we should keep this linter rule.

I would love it if we could restrict the css we write as much as possible.
Then it would not be that hard to follow new standards.

I would not refactor the components now, but get as soon as possible to a state where a linter can at least tell us about some not ideal practises.

**Comment by BacLuc** [comment]
And as far as i remember, vue3 did not like not scoped style tags, thus i would rather fix that now than with the big bang vue3 upgrade

File: frontend/package.json [hunk 4]
```diff
@@ -145,17 +145,18 @@
           ]
         }
       ],
+      "vue-scoped-css/enforce-style-type": "error",
```
Comment (manuelmeister) [inline frontend/package.json:148]:
```suggestion
      "vue-scoped-css/enforce-style-type": "warn",
```

File: frontend/src/App.vue [hunk 5]
```diff
@@ -136,6 +138,8 @@ export default {
 </style>
 
 <style scoped>
+/* <v-footer> is transformed to <footer class="v-footer"> */
+/* eslint-disable-next-line vue-scoped-css/no-unused-selector */
```
Comment (manuelmeister) [inline frontend/src/App.vue:142]: I hate that we need to disable the rule for a line. I'd rather remove this eslint rule. This gives us a false sense of security.

**Review by manuelmeister — CHANGES_REQUESTED** [review]
> And as far as i remember, vue3 did not like not scoped style tags, thus i would rather fix that now than with the big bang vue3 upgrade

I don't think so. It isn't even in the official eslint rules. I agree that it is better to use scoped for 80% of the styles, but for the rest it doesn't just work seamless.

I think, I'd rather fix these issues with the style refactoring. So I can ensure that the design is not broken.

**Comment by BacLuc** [comment]
What i want to omit is the following:

We have it somewhat running, but we have thousands of these warnings:

image

and we don't know how we want to continue.
Is this ok, is it not ok?

thus i want to remove as much issues as possible.

I don't know how you want to ensure that everything looks good again with refactoring.
I would say we are faster if we just check if we can scope the css, and else we document, why it cannot be scoped.

This will also help with the refactoring.
